backend/fms_core/models/sample.py

```diff
@@ -264,65 +127,27 @@ def add_error(field: str, error: str):
 
         self.normalize()
 
-        # Sample Kind
-        sample_kind_choices = (Sample.BIOSPECIMEN_TYPE_NA_CHOICES if self.extracted_from
-                               else Sample.BIOSPECIMEN_TYPE_CHOICES)
-
-        if self.sample_kind_id is not None:
-            if self.sample_kind.name not in frozenset(c[0] for c in sample_kind_choices):
-                add_error(
-                    "sample_kind",
-                    (f"Sample Kind name {self.sample_kind.name} not valid for "
-                     f"{' extracted' if self.extracted_from else ''} sample {self.name}"),
-                )
-            # Check concentration fields given sample_kind
-            if self.sample_kind.name in Sample.BIOSPECIMEN_TYPES_CONC_REQUIRED and self.concentration is None:
-                add_error("concentration", "Concentration must be specified if the sample_kind is DNA")
-
-            # Check tissue source given extracted_from
-
-            if self.tissue_source and self.sample_kind.name not in Sample.BIOSPECIMEN_TYPES_NA:
-                add_error("tissue_source", "Tissue source can only be specified for a nucleic acid sample.")
-
-            if self.transferred_from and self.sample_kind != self.transferred_from.sample_kind:
-                    add_error("sample_kind", "Sample kind need to remain the same during transfer")
+        DerivedSample = apps.get_model("fms_core", "DerivedSample")
+        # Check concentration fields given sample_kind
+        if self.concentration is None and (self.is_pool or self.derived_samples.objects.first().sample_kind.name in DerivedSample.BIOSPECIMEN_TYPES_CONC_REQUIRED):
```

Comment:
self.derived_samples.objects.first() ? 

You want to get the only one DerivedSample of a sample? Rather than calling this manually there should be a property that gets you the wanted DerivedSample. We should avoid those ... objects.first() directly in the code

Reply:
hehe I disagree, this is better abstracted into a single function, you could simply raise an exception if there are more than 1 derived_samples for a sample.

What about we create a custom exception type for operations attempted on pools, that should not be done on a pool (such as extraction for example). That'd help us with the coding

Reply:
I would like it in a function property but i would be useless as far as i can see. The idea of an exception (validationError) for wrong operation is not too bad but the issue here is that mostly these are not bad operation for Pools they are just workaround that are specific to some cases. It is ok to get the first derivedsample in a pool ... it is wrong to think it is the only one though. The logic i did here assume it is not a pool but i would have to cover the case of the pool too if it was not an extraction (for a transfer for exemple). Give me an example maybe i do not get your idea.

Reply:
yes if I understand it correctly you mention them being 'workaround that are specific to some cases'. the workaround is using 'first()' to denote fetching the only one derived_sample attached to a sample if the sample is not a pool. I prefer the idea you suggested of having the derived_sample_when_not_pool function that tests that it is indeed not a pool, but of course we can have a shorter/more concise function (should be a property imo) name

Reply:
ok i'll make a shortcut property for that case ... we'll see if it's used more than once ;).